gpu-type option

Gldkslfmsd opened the pull request:
option `--gpu-type.` to select GPU type by name, such as a40, NVIDIA RTX 3090, etc. 

It translates it into slurm `--constraint=gpuram48G,gpu_cc8.6` option.

Gldkslfmsd committed:
gpu-type option

maybe the code could be cleaner...

tuetschek commented:
So if 2 GPU types have the same memory and CC, they're indistinguishable for this approach, right? Maybe we can provide specific nodes list to `qsub` instead? 

Otherwise, it looks god to me, I'd be happy to merge this for now.

Gldkslfmsd commented:
> So if 2 GPU types have the same memory and CC, they're indistinguishable for this approach, right?

yes, but currently it's not

> Maybe we can provide specific nodes list to `qsub` instead

how? I don't know any better way

> 
> Otherwise, it looks god to me, I'd be happy to merge this for now.

thanks. Documentation is pending, as far as I know

tuetschek commented:
>>    Maybe we can provide specific nodes list to `qsub` instead

> how? I don't know any better way

`qsub [...] --nodelist=dll-8gpu4,dll-8gpu5` works, just tested it.

Gldkslfmsd committed:
more effective qruncmd:

- don't start all workers but give a chance to the submitting thread
- don't read and print the whole job_*.out at once but in chunks, which
  gives a chance for execution to the submitting thread
